library/threading.c

```diff
@@ -35,12 +36,17 @@ static void threading_mutex_init_pthread( mbedtls_threading_mutex_t *mutex )
     if( mutex == NULL )
         return;
 
-    mutex->is_valid = pthread_mutex_init( &mutex->mutex, NULL ) == 0;
+    mutex->is_valid = (pthread_mutex_init( &mutex->mutex, NULL ) == 0) ? 1 : 0;
+}
+
+static bool is_mutex_valid(mbedtls_threading_mutex_t *mutex)
```

Comment:
why not change the return value to `char` and just return mutex->is_valid?

Reply:
style: add space after `(` and before `)`

Reply:
This is almost what @RonEld said, but I think uniformity with the rest of the library would be achieved by returning `int` here.

Reply:
Fixing both.

Reply:
> Style: As far as I remember, we rather use the naming scheme `xxx_is_yyy()` as opposed to `is_xxx_yyy()`

I disagree @hanno-arm. grepping where we use the `xxx_is_yyy` naming convention, it's mainly in variables. Where we do use it in functions it follows the convention of `thing_is_different_thing`, where there's a relationship between the two.

Here, we're testing a state or quality of something, where the `is_xxx_yyy()` idiom is very widely accepted and more natural. So I would have said `is_xxx_yyy()` is preferable to `is_xxx_yyy()`.

This is all academic and it's too minor to merit changing, but I think it's worth making the point.
(Please don't waste time reverting the change).

Reply:
@sbutcher-arm Given that there are no functions of the form `is_xxx` in the library, but numerous of the form `xxx_is_yyy` (leaving any further details aside), I still think the the chosen naming is the preferred one - but as you say, it's probably not worth discussing further, so as long as you don't reject the PR because of this, let's leave it.